backend/src/services/note.ts

```diff
@@ -0,0 +1,107 @@
+import { assert } from "console";
+import { createHash } from "crypto";
+import preSessionQuestions from "../models/preQuestionsList.json";
+import postSessionQuestions from "../models/postQuestionsList.json";
+import { Note } from "../models/notes";
+
+/*
+ * Class definition for an Answer to a question, can either be textbox or bullet boxes.
+ */
+class Answer {
+  answer: string | Array<string>;
+
+  type: string;
+
+  id: string; // hashed from question
+
+  constructor(type: string, id: string) {
+    this.answer = "";
+    this.type = type;
+    this.id = id;
+  }
+
+  /**
+   * Sets the text answer of an answer object. If the answer is a textbox, the text will be set to the input.
+   * Otherwise, the text will be added to the answer ArrayList.
+   * @param input: Text of answer to input
+   */
+  setAnswer(input: string): void {
+    if (typeof this.answer === "string") {
+      this.answer = input;
+    } else {
+      assert(Array.isArray(this.answer));
```

Comment:
Good job with these asserts! What happens when the assert fail? Ideally if the assert fails, our catchall error handler should pick this up and report as internal error to client.